mcstatus/tests/test_motd.py

```diff
@@ -0,0 +1,212 @@
+                {"strikethrough": True, "color": "#f9daff", "text": "="},
+                {"strikethrough": True, "color": "#ffd9ff", "text": "="},
+                {"color": "white", "text": " "},
+                {"bold": True, "color": "#66ff99", "text": "C"},
+                {"bold": True, "color": "#75f5a2", "text": "r"},
+                {"bold": True, "color": "#84ebab", "text": "e"},
+                {"bold": True, "color": "#93e2b4", "text": "a"},
+                {"bold": True, "color": "#a3d8bd", "text": "t"},
+                {"bold": True, "color": "#b2cfc6", "text": "i"},
+                {"bold": True, "color": "#c1c5cf", "text": "v"},
+                {"bold": True, "color": "#d1bbd8", "text": "e"},
+                {"bold": True, "color": "#e0b2e1", "text": "F"},
+                {"bold": True, "color": "#efa8ea", "text": "u"},
+                {"bold": True, "color": "#ff9ff4", "text": "n "},
+                {"strikethrough": True, "color": "#b3eeff", "text": "="},
+                {"strikethrough": True, "color": "#b9ecff", "text": "="},
+                {"strikethrough": True, "color": "#c0eaff", "text": "="},
+                {"strikethrough": True, "color": "#c7e8ff", "text": "="},
+                {"strikethrough": True, "color": "#cee6ff", "text": "="},
+                {"strikethrough": True, "color": "#d5e4ff", "text": "="},
+                {"strikethrough": True, "color": "#dce2ff", "text": "="},
+                {"strikethrough": True, "color": "#e3e0ff", "text": "="},
+                {"strikethrough": True, "color": "#eadeff", "text": "="},
+                {"strikethrough": True, "color": "#f1dcff", "text": "="},
+                {"strikethrough": True, "color": "#f8daff", "text": "="},
+                {"strikethrough": True, "color": "#ffd9ff", "text": "="},
+                {"strikethrough": True, "color": "#f4dcff", "text": "="},
+                {"strikethrough": True, "color": "#f9daff", "text": "="},
+                {"strikethrough": True, "color": "#ffd9ff", "text": "="},
+                {"color": "white", "text": " \n "},
+                {"bold": True, "color": "#E5E5E5", "text": "The server has been updated to "},
+                {"bold": True, "color": "#97ABFF", "text": "1.17.1"},
+            ],
+            "text": "",
+        }
+
+    def test_correct_result(self, source):
+        assert Motd.parse(source) == Motd(
+            [
+                # fmt: off
+                " ",
+                RawHTMLColor(hex="#b3eeff", rgb=(179, 238, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                RawHTMLColor(hex="#b9ecff", rgb=(185, 236, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                RawHTMLColor(hex="#c0eaff", rgb=(192, 234, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
+                RawHTMLColor(hex="#c7e8ff", rgb=(199, 232, 255)), Formatting.STRIKETHROUGH, "=", Formatting.RESET,
```

Comment:
Thought about remake it as minecraft does. Create `MotdElement` with fields `color`, `formatting`, `text` etc. and every item in `Motd.raw` will not ends with `Formatting.RESET`.

But this will introduce more complex parsing in `plain` and `html` properties, which will anyway add `Formatting.RESET` after every item from `Motd.raw`.

Reply:
I think that's a good idea, if the parsing in `plain` and `html` is that complex, we could easily introduce caching to those properties

Reply:
No, they fast. Just complex to read.

Reply:
This introduce a lot of problems in `html`, `ansi` etc. parsers.